library/core/src/alloc/layout.rs

```diff
@@ -492,3 +499,26 @@ impl fmt::Display for LayoutError {
         f.write_str("invalid parameters to Layout::from_size_align")
     }
 }
+
+#[cfg(kani)]
+#[unstable(feature="kani", issue="none")]
+mod verify {
+    use super::*;
+
+    #[kani::proof_for_contract(Layout::from_size_align_unchecked)]
+    pub fn check_from_size_align_unchecked() {
+        let shift_index = kani::any::<usize>();
+        kani::assume(shift_index < core::mem::size_of::<usize>() * 8);
+        let a : usize = 1 << shift_index;
+        kani::assume(a > 0);
+
+        let s = kani::any::<usize>();
+        kani::assume(s <= isize::MAX as usize - (a - 1));
```

Comment:
Why are you adding these assumptions here? We are going to assume this in the preconditions.

Reply:
Oh, indeed, misconception on my part. Fixed.